dht-node/src/dht_node/index.ts

```diff
@@ -184,3 +187,35 @@ export const startDHT = async (topic: string): Promise<void> => {
     logger.error('DHT unexpected error:', err)
   }
 }
+
+async function writeHomeCommunityEnries(pubKey: any): Promise<CommunityApi[]> {
+  const homeApiVersions: CommunityApi[] = Object.values(ApiVersionType).map(function (apiEnum) {
+    const port =
+      Number.parseInt(CONFIG.FEDERATION_COMMUNITY_API_PORT) + Number(apiEnum.replace('_', ''))
+    const comApi: CommunityApi = {
+      api: apiEnum,
+      url: CONFIG.FEDERATION_COMMUNITY_URL + ':' + port.toString() + '/api/',
+    }
+    return comApi
+  })
+  try {
+    // first remove privious existing homeCommunity entries
+    DbCommunity.createQueryBuilder().delete().where({ foreign: false }).execute()
+
+    homeApiVersions.forEach(async function (homeApi) {
+      const homeCom = new DbCommunity()
+      homeCom.foreign = false
+      homeCom.apiVersion = homeApi.api
+      homeCom.endPoint = homeApi.url
+      homeCom.publicKey = pubKey.toString('hex')
+
+      // this will NOT update the updatedAt column, to distingue between a normal update and the last announcement
+      await DbCommunity.insert(homeCom)
+      logger.info(`federation home-community inserted successfully: ${JSON.stringify(homeCom)}`)
+    })
+  } catch (err) {
+    throw new Error(`Federation: Error writing HomeCommunity-Entries: ${err}`)
+  }
+
+  return homeApiVersions
```

Comment:
```suggestion
```

Reply:
the signature of this methode needs a return value of type Promise<CommunityApi[]>
if the return statement is shifted in the try-catch-statement the compiler shows missing return value error